lib/manageiq/providers/workflows/builtin_methods.rb

```diff
@@ -25,17 +43,29 @@ def self.email(params = {}, _secrets = {}, _context = {})
             if runner_context["success"]
               runner_context["output"] = miq_task.message
             else
-              error!(runner_context, :cause => miq_task.message)
+              BuiltinRunner.error!(runner_context, :cause => miq_task.message)
             end
           end
 
           runner_context
         end
 
-        private_class_method def self.error!(runner_context = {}, cause:, error: "States.TaskFailed")
-          runner_context.merge!(
-            "running" => false, "success" => false, "output" => {"Error" => error, "Cause" => cause}
-          )
+        private_class_method def self.miq_request_task_status!(runner_context)
+          miq_request_task = ::MiqRequestTask.find_by(:id => runner_context["miq_request_task_id"])
+
+          case miq_request_task&.statemachine_task_status
+          when nil
+            reason = "Unable to find MiqRequestTask id: [#{runner_context["miq_request_task_id"]}]"
+            BuiltinRunner.error!(runner_context, :cause => reason)
+          when "error"
+            reason = request_task.message&.sub(/^Error: /, "")
+            BuiltinRunner.error!(runner_context, :cause => reason)
+          when "retry"
+            runner_context["running"] = true
+            runner_context
+          when "ok"
+            BuiltinRunner.success!(runner_context, :output => {"Result" => "provisioned"})
```

Comment:
We should probably return something that subsequent states could make use of, maybe the resulting `MiqRequestTask#destination` which is a `Service`

Example use case, run some automation against the vm(s) that get created.